logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -99,6 +102,23 @@ def config_init(params)
       end
     end
 
+    # Resolve environment variables references
+    params.each do |name, value|
+      if (value.is_a?(Hash))
+        value.each do |valueHashKey, valueHashValue|
+          value[valueHashKey.to_s] = replace_env_placeholders(valueHashValue)
```

Comment:
The only reason was because in `mixin.rb`, there are numerous `name.to_s` while processing params (ex: line 87).
But I agree that for a hash key, there's probably no need to do that.

Tell me if you want I remove it.